doc/Requirements.md

```diff
@@ -0,0 +1,184 @@
+
+A syncer MUST support retrieving the mapped parameters value from the mapped hardware.
+(Why: to retrieve a parameter value at the start of the PF.)
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in it's containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+
+Syncer are dedicated to map one hardware. Multiple syncers that can talk
+Plugins contain subsystem libraries.
+A subsystem library MUST
+
+## Mapping
+One or more
+
+# Domains
+
+## Selection criterion
+
+A selection criterion MUST have one, and only one, state at a given time.
+
+A selection criterion SHOULD know at construction all possible states.
+(Why: To be able to validate:
+       - rules on start
+       - state changes)
+
+The selection criterion possible states SHOULD be specifiable by at least:
+ - Directly a state set (Input -> states == identity)
+   (Note: called exclusive criterion)
+ - A combination of values
+   (Note: combination in the mathematical sense “ab” -> [“”, “a”, “b”, “ab”])
+   (Note: called inclusive criterion)
+
+Parameter values MUST be able to change based on selection criterion state.
+Note: this does not mean that a function can be defined between both states.
+      Ie: Parameter values MAY depend on previous selection criterion states.
+
+ 
+### Definitions
+ - Selection criterion rule: function (in the mathematical sense) that MUST given a selection criterions return a Boolean.
+
+ - Rule: an Boolean expression of Selection criterion rules
+   (Note: implementation only allows AND and OR combinaison)
+
+### Rules
+
+One MUST always be able to create a selection criterion rule from a given selection criterion
+(Why: if no rules can be formulated from a criterion, it is useless)
```

Comment:
If so, it adds a requirement on criteria: they must have at least one valid state.